pkg/mediationcontainer/client_protobuf_endpoint.go

```diff
@@ -168,18 +174,20 @@ func (messageWaiter *SingleMessageWaiter) getMessage(endpoint ProtobufEndpoint)
 
 	// Parse the input stream using the registered message handler
 	messageHandler := endpoint.GetMessageHandler()
-	err := messageHandler.parse(rawBytes)
+	parsedMsg, err := messageHandler.parse(rawBytes)
 	if err != nil {
 
```

Comment:
should we do sth when there is an error? Otherwise a parsedMsg (= nil) will be passed into the channel.